compiler/optimizer/VPHandlers.cpp

```diff
@@ -93,6 +93,42 @@
 extern TR::Node *constrainChildren(OMR::ValuePropagation *vp, TR::Node *node);
 extern TR::Node *constrainVcall(OMR::ValuePropagation *vp, TR::Node *node);
 
+static bool canTrustAsFixedClass(OMR::ValuePropagation *vp, TR::SymbolReference *symRef, TR_OpaqueClassBlock *classObject)
+   {
+#ifdef J9_PROJECT_SPECIFIC
+   if (!TR::Compiler->om.areFlattenableValueTypesEnabled())
+      return true;
+
+   if ((classObject == NULL) && symRef && symRef->getSymbol()->isClassObject())
```

Comment:
May I suggest making these tests for NULL/non-NULL consistent?  Either `!classObject && symRef` or `(classObject == NULL) && (symRef == NULL)`